db/indexes.go

```diff
@@ -363,18 +363,26 @@ func waitForIndex(bucket base.N1QLStore, indexName string, queryStatement string
 	retrySleeper := base.CreateMaxDoublingSleeperFunc(180, 100, 5000)
 	retryCount := 0
 	for {
-		_, err := bucket.Query(queryStatement, nil, base.RequestPlus, true)
-		if err == nil {
+		resultSet, resultsError := bucket.Query(queryStatement, nil, base.RequestPlus, true)
+
+		// Immediately close results. We don't need these
+		resultSetCloseError := resultSet.Close()
```

Comment:
Do we need a nil check on resultSet here?  collection.Query looks like it's going to return nil in some error cases, at least.